test/test.c

```diff
@@ -16311,6 +16311,11 @@ test_ServerPoolUpdatedOnClusterUpdate(void)
         const char *urls[] = {"127.0.0.1:4222", "127.0.0.1:4223", "127.0.0.1:4224"};
         test("Check pool: ");
         s = _checkPool(conn, (char**)urls, (int)(sizeof(urls)/sizeof(char*)));
+        if (s != NATS_OK)
+        {
+            nats_Sleep(100);
```

Comment:
@kozlovic if you don't mind a quick look here to confirm.

I think the the (connected) server was sending 2 INFO packets, the first was coming with obsolete list of URL, and then the second one would correct it. Not sure if there's a better way to wait for the second info packet to arrive in the test, but this seems to have fixed the flapping.

Reply:
I am not able to reproduce, but _checkPool() is already sleeping and waiting until it gets the expected array, but I realize that we don't retry for the expected size (just the expected content). We could improve _checkPool instead of adding the sleep here...

Reply:
@kozlovic Thanks! please https://github.com/nats-io/nats.c/pull/680/commits/209e7ac27c9ef0e98f7b53b972fad2f046c1c532